app/logic/emailHandler.py

```diff
@@ -0,0 +1,84 @@
+    def __init__(self, mail, emailInfo):
+        app.config.update(
+            MAIL_SERVER=app.config['mail']['server'],
+            MAIL_PORT=app.config['mail']['port'],
+            MAIL_USERNAME= app.config['mail']['admin_username'],
+            MAIL_PASSWORD= app.config['mail']['admin_password'],
+            REPLY_TO_ADDRESS= app.config['mail']['reply_to_address'],
+            MAIL_USE_TLS=app.config['mail']['tls'],
+            MAIL_USE_SSL=app.config['mail']['ssl'],
+            MAIL_DEFAULT_SENDER=app.config['mail']['default_sender'],
+            #ALWAYS_SEND_MAIL=default['ALWAYS_SEND_MAIL']
+            MAIL_OVERRIDE_ALL=app.config['mail']['override_addr']
+        )
+        self.emailInfo = emailInfo
+        self.mail = mail
+        self.mail.connect()
+        # print(app.config['MA'])
+    def updateSenderEmail(self, message):
+        """ Updates who is sending the emails based on the event_list form. """
+
+        if self.emailInfo['emailSender'] == 'CELTS Admins':
+            message.sender = app.config['mail']['admin_username']
+            # app.config.update(
+            #     mail_username= app.config['mail']['admin_username'],
+            #     mail_password = app.config['mail']['admin_password']
+            # )
+            # print("\n\n"+app.config["mail_username"])
+
+        elif self.emailInfo['emailSender'] == 'CELTS Student Staff':
+            app.config.update(
+                mail_username = app.config['mail']['staff_username'],
+                mail_password = app.config['mail']['staff_password']
+            )
+
+    def sendEmail(self, message: Message, emails):
+        """ Updates the sender and sends the email. """
+        # self.updateSenderEmail(message)
+        # print(f'\n{message.sender},\n{message.subject}')
+        # if '@' in self.emailInfo['emailSender']: #if the current user is sending the email
+        #     msg = self.emailInfo['message'].replace(" ",'%20')
+        #     subject = self.emailInfo['subject'].replace(" ",'%20')
+        #     if app.config['mail']['override_addr']:
+        #         recipients = app.config['mail']['override_addr']
+        #     else:
+        #         recipients = emails
+        #     webbrowser.open_new(f'mailto:?to={recipients}&subject={subject}&body={msg}')
+        if 'sendIndividually' in self.emailInfo:
+            if app.config['mail']['override_addr']:
+                message.recipients = [app.config['mail']['override_addr']]
+            with self.mail.connect() as conn:
+                for email in emails:
+                    message.recipients = [email]
+                    conn.send(message)
+        else:
+            if app.config['mail']['override_addr']:
+                message.recipients = [app.config['mail']['override_addr']]
+
+            message.reply_to = app.config['mail']["reply_to_address"]
+            self.mail.send(message)
+        return 1

```

Comment:
We don't need to be returning integers. Let's use True

Reply:
Should you check the return value of the actual message sending to make sure it succeeded? I'm not sure what the possible return values are.

Reply:
This is also a good method to have a try/catch, so that you can return False if there was an exception in email sending. That way the calling code in the controller only has to concern itself with a true or false outcome, not true, false, and exception

Reply:
``mail.send`` doesn't seem to have a return value. I added a try except block and printed out the error. The function will return false if it does run into an error or just return true if there are no errors.